### PR TITLE
Allow single NOT operator in Table Query

### PR DESCRIPTION
**Pull request by edwin-huber**
Addresses issues in the query parser, specifically allowing a single NOT operator.
Updates the unit and integration tests

Fixes: #1828

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -19,8 +19,12 @@ Blob:
 Table:
 
 - Fixed issue for querying on identifiers starting with underscore.
+<<<<<<< HEAD
+- Corrected query parsing logic for single boolean terms.
+=======
```
Comment (blueww) [inline ChangeLog.md:24]: change log need merge.

Comment (edwin-huber) [reply ChangeLog.md:24]: 🙈 thanks!

Comment (blueww) [reply ChangeLog.md]: As we just have a new release, it looks the change log is not based on the latest release, so the change log location is not right.
Please rebase on latest main.

**Review by blueww — APPROVED** [review]

**Comment by PrasantJillella** [comment]
@EmmaZhu 
@XiaoningLiu 
please take a look once you get some time. Thank you.

**Pull request merged**